internal/context/init.go

```diff
@@ -81,77 +87,49 @@ func promptAllModules(modules map[string]moduleconfig.ModuleConfig) map[string]s
 	parameterValues := make(map[string]string)
 	for _, config := range modules {
 		var err error
-		parameterValues, err = module.PromptParams(config, parameterValues)
+		parameterValues, err = PromptModuleParams(config, parameterValues)
 		if err != nil {
 			exit.Fatal("Exiting prompt:  %v\n", err)
 		}
 	}
 	return parameterValues
 }
 
-// global configs
-func promptPushRepoUpstream() string {
-	providerPrompt := promptui.Prompt{
-		Label:     "Should the created projects be checked into github automatically? (y/n)",
-		Default:   "y",
-		AllowEdit: false,
-	}
-	providerResult, err := providerPrompt.Run()
-	if err != nil {
-		exit.Fatal("Exiting prompt:  %v\n", err)
-	}
-	return providerResult
-}
-
-func promptGithubRootOrg() string {
-	providerPrompt := promptui.Prompt{
-		Label:     "What's the root of the github org to create repositories in?",
-		Default:   "github.com/",
-		AllowEdit: true,
-	}
-	result, err := providerPrompt.Run()
-	if err != nil {
-		exit.Fatal("Exiting prompt:  %v\n", err)
-	}
-	return result
-}
-
-func promptGithubPersonalToken(projectName string) string {
-	defaultToken := ""
-
-	project := globalconfig.GetUserCredentials(projectName)
-	if project.GithubResourceConfig.AccessToken != "" {
-		defaultToken = project.GithubResourceConfig.AccessToken
-	}
-
-	providerPrompt := promptui.Prompt{
-		Label:   "Github Personal Access Token with access to the above organization",
-		Default: defaultToken,
-	}
-	result, err := providerPrompt.Run()
-	if err != nil {
-		exit.Fatal("Prompt failed %v\n", err)
-	}
-
-	// If its different from saved token, update it
-	if project.GithubResourceConfig.AccessToken != result {
-		project.GithubResourceConfig.AccessToken = result
-		globalconfig.Save(project)
-	}
-	return result
-}
-
-func promptProjectName(projectName string) string {
-	providerPrompt := promptui.Prompt{
-		Label:     "Project Name",
-		Default:   projectName,
-		AllowEdit: false,
-	}
-	result, err := providerPrompt.Run()
-	if err != nil {
-		exit.Fatal("Prompt failed %v\n", err)
+func getProjectPrompts(projectName string) map[string]PromptHandler {
+	return map[string]PromptHandler{
+		"projectName": {
```

Comment:
Since it was supplied on the command line we probably don't need to prompt for name. Or maybe we make it optional on the command line and then conditionally prompt here?

Reply:
I'll do this as part of my ticket.

Reply:
https://github.com/commitdev/zero/pull/153 yeah we removed the projectName as a positional argument altogether just a little earlier, ill need to rebase with that change